tools/bin/ci_credentials.sh

```diff
@@ -1,26 +1,150 @@
 #!/usr/bin/env bash
 
 . tools/lib/lib.sh
+. tools/lib/gcp-token.sh
 
 set -e
 
+COMMAND_NAME=$1
+
+# all secrets will be loaded if the second argument is not present
+CONNECTOR_FULLNAME=${2:-all}
+CONNECTOR_NAME=`echo ${CONNECTOR_FULLNAME} | rev | cut -d'/' -f1 | rev`
+
+GSM_SCOPES="https://www.googleapis.com/auth/cloud-platform"
+
+declare -A SECRET_MAP
+
+
 function write_standard_creds() {
```

Comment:
suggest renaming to `read_secrets`

Reply:
@sherifnada,
this is the single debatable function name. this function is used for the old logic (that huge mapping list for secrets).
Sure, we can rename it but contributors will have some concern if they don't read updated secrets' flow docs before.
Thus I saved its old name.

Reply:
I think devs will be able to pick up on what's happening here; renaming will pose a problem. It's better that we have a slight confusion that accurately reflects the current/future state rather than stay beholden to decisions from the past :)

Reply:
ok, but I will rename it immediately before merging to decrease conflicts' amount